### DISCUSSION TITLE
Clerk Integration with next 13.1.6  Missing publishableKey.

### DISCUSSION THREAD

**Discussion by humancodex** [clerk/javascript#3131] [Support request - DO NOT OPEN HERE]
I want to address some serious issues we are encountering in the production deployment of our solution based on Next.js, version "13.1.6". When attempting to access the page "/Admin", we encounter the following error:

Error occurred prerendering page "/Admin". Read more: https://nextjs.org/docs/messages/prerender-error
Error: @clerk/nextjs: Missing publishableKey. You can get your key at https://dashboard.clerk.com/last-active?path=api-keys.

Error occurred prerendering page "/marketplace". Read more: https://nextjs.org/docs/messages/prerender-error
Error: @clerk/nextjs: Missing publishableKey. You can get your key at https://dashboard.clerk.com/last-active?path=api-keys.

_app.js 
<ClerkProvider {...pageProps} publishableKey={process.env.NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY}> <NFTProvider> <Component {...pageProps} /> </NFTProvider> </ClerkProvider>
It's important to note that the Clerk publishable key is fetched from the environment variables. Specifically, it should be available as NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY.

The project structure :

![Captura de pantalla 2024-04-04 205510](https://github.com/clerk/javascript/assets/85398460/c4d8132e-c98a-41e6-a70d-321439857477)

We are deploying on vercel

Upon further analysis, we have identified that the error lies in the absence of a publishableKey required by Clerk for the proper functioning of our application. This key is essential for authenticating and authorizing requests with Clerk.

**Comment by jescalan** [comment]
https://github.com/orgs/clerk/discussions/3829
